collections/distinct.ts

```diff
@@ -0,0 +1,18 @@
+// Copyright 2018-2021 the Deno authors. All rights reserved. MIT license.
+
+/**
+ * Returns all distinct elements in the given array, preserving order by first occurence
+ *
+ * Example:
+ *
+ * ```typescript
+ * const numbers = [ 3, 2, 5, 2, 5 ]
+ * const distinctNumbers = distinct(numbers)
+ *
+ * console.assert(distinctNumbers === [ 3, 2, 5 ])
```

Comment:
nitpick: `console.assert` will always fail because their identities are different. `assert.deepStrictEqual(distinctNumbers, [3,2,5])` in nodejs or `expect(distinctNumbers).toEqual([3,2,5])` in jest might be a better choice.

Reply:
You are absolutely right - we should make sure that those examples actually work. I just totally missed that. I will put it on my to do list to do a sweep of that in the end.